### ISSUE TITLE
Prevent websites from hijacking built-in keyboard shortcuts

### ISSUE THREAD

**Issue by wknapik** [brave/brave-browser#14948]
## Description 

Some sites hijack the shortcut for "search in page". I hit ^F and instead of the native browser search function, I get whatever the site author decided I should see instead.

## Steps to Reproduce

   1. Go to https://forum.snapcraft.io/t/snap-connections-command/4296
   2. Click anywhere on the page, to put the focus there
   3. Hit ^F to search
   4. The browser search does not work, instead a site-specific search pops up

## Actual result:

A keyboard shortcut for a bult-in feature has been hijacked.

## Expected result:
Use built-in search.

## Reproduces how often: 

Easily reproduced.

## Brave version (brave://version info)

Brave | 1.22.66 Chromium: 89.0.4389.90 (Official Build) (64-bit)
-- | --
Revision | 62eb262cdaae9ef819aadd778193781455ec7a49-refs/branch-heads/4389@{#1534}
OS | Linux

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Does the issue resolve itself when disabling Brave Rewards? No
- Is the issue reproducible on the latest version of Chrome? Yes

**Comment by antonok-edm** [comment]
Most sites that hijack Ctrl+F make it so that hitting Ctrl+F a second time will bring up the browser search - but this does have to be specifically implemented by the page and is not a feature of the browser.

**Comment by dentistformyeye** [comment]
> Most sites that hijack Ctrl+F make it so that hitting Ctrl+F a second time will bring up the browser search - but this does have to be specifically implemented by the page and is not a feature of the browser.

In addition, Ctrl+G also triggers the browser's built-in search feature and is less likely to be hijacked.

> Prevent websites from hijacking built-in keyboard shortcuts

I'd like for this to be something I can easily switch off temporarily as opposed to a flag (which would require a browser-restart to toggle).

**Comment by ryanbr** [comment]
Using `uBO Annoyances` via `brave://adblock`, they include some site-specific mitigations. 

* Disable Right click
* Disable Dev tools access
* Disable Mouse selection

Example for:
`hollywoodmask.com` https://github.com/uBlockOrigin/uAssets/commit/56804ca
`scrolller.com` https://github.com/uBlockOrigin/uAssets/commit/2bd0490

Whether we include these fixes in our default lists, that's not my call here.

**Comment by wknapik** [comment]
Just ran into a page that overrides ctrl+shift+pgup/pgdown, which moves a tab left/right.
Sites should not be able to hijack basic browser UI controls.

**Comment by Aeroverra** [comment]
Agreed, make it a toggle if you are unsure about it. I think a lot of power users would agree this is something that is long overdue.
